src/control/PlanarArmController.h

```diff
@@ -149,13 +150,19 @@ class PlanarArmController {
 		double radius = kin.getSegLens().sum() * safetyFactor;
 
 		if (eePos.norm() > radius) {
-			// TODO: will need to eventually shrink velocity vector until it is within radius
-			// instead of just normalizing it.
-
-			eePos.normalize();
-			eePos *= radius;
+			double radius = kin.getSegLens().sum();
+			// new position is outside of bounds. Set new EE setpoint so it will follow the velocity
+			// vector instead of drifting along the radius.
+			// setpoint = setpoint inside circle
+			// eePos = new setpoint outside circle
```

Comment:
I think we need a bit more math explanation. I'd say just state the equation we solve after these definitions, so at least readers can re-derive the solution if they want.

Reply:
I think this explanation could still use some work. Like I mentioned, put the original equation we started with, which will allow others to rederive the solution if necessary.